src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -88,55 +84,55 @@ private string TryWriteSonarLintXmlFile(IEnumerable<SonarRule> activeRules)
         }
         else
         {
-            var content = RoslynSonarLint.GenerateXml(activeRules, sonarProperties, language);
+            var content = RoslynSonarLint.GenerateXml(rules.Where(x => x.IsActive), sonarProperties, language);
             logger.LogDebug(Resources.RAP_WritingAdditionalFile, sonarLintXmlPath);
             File.WriteAllText(sonarLintXmlPath, content);
             return sonarLintXmlPath;
         }
     }
 
-    private IEnumerable<AnalyzerPlugin> FetchAnalyzerPlugins(IEnumerable<SonarRule> activeRules)
+    private IEnumerable<AnalyzerPlugin> FetchAnalyzerPlugins()
     {
-        var partialRepoKeys = ActiveRulesPartialRepoKeys(activeRules);
-        List<Plugin> plugins = new();
-        foreach (var partialRepoKey in partialRepoKeys)
+        var plugins = CreatePlugins();
+        if (plugins.Length > 0)
         {
-            if (sonarProperties.TryGetValue($"{partialRepoKey}.pluginKey", out var pluginKey)
-                && sonarProperties.TryGetValue($"{partialRepoKey}.pluginVersion", out var pluginVersion)
-                && sonarProperties.TryGetValue($"{partialRepoKey}.staticResourceName", out var staticResourceName))
-            {
-                plugins.Add(new Plugin(pluginKey, pluginVersion, staticResourceName));
-            }
-            else if (!partialRepoKey.StartsWith(LegacyServerPropertyPrefix))
-            {
-                logger.LogInfo(Resources.RAP_NoAssembliesForRepo, partialRepoKey, language);
-            }
+            logger.LogInfo(Resources.RAP_ProvisioningAnalyzerAssemblies, language);
```

Comment:
Where are UTs to assert these if/else log messages?

The logs should be ideally asserted in existing UTs